src/lib/certmap/sss_certmap_ldap_mapping.c

```diff
@@ -60,7 +71,226 @@ struct template_table template_table[] = {
     {"subject_principal", name_attr, empty},
     {NULL, NULL, NULL}};
 
+struct template_table template_table_u1[] = {
+    {"serial_number", empty, dec_hex_conv},
+    {"subject_key_id", empty, hex_conv},
+    {"cert", empty, digest_conv},
+    {"subject_dn_component", attr_name_and_or_number, empty},
+    {"issuer_dn_component", attr_name_and_or_number, empty},
+    {"sid", sid_rid_attr, empty},
+    {NULL, NULL, NULL}};
+
+int check_attr_name_and_or_number(TALLOC_CTX *mem_ctx, const char *inp,
+                                  char **_attr_name, int32_t *_number)
+{
+    int ret;
+    char *sep;
+    char *end;
+    char *endptr = NULL;
+    char *attr_name = NULL;
+    int32_t number = 0;
+
+    if (inp == NULL) {
+        attr_name = NULL;
+        number = 0;
+        ret = 0;
+        goto done;
+    }
+
+    sep = strchr(inp, '[');
+    if (sep == NULL) {
+        attr_name = talloc_strdup(mem_ctx, inp);
+        if (attr_name == NULL) {
+            return ENOMEM;
+        }
+        number = 0;
+    } else {
+        end = strchr(sep, ']');
+        if (end == NULL || end == (sep + 1) || *(end + 1) != '\0') {
+            return EINVAL;
+        }
+        errno = 0;
```

Comment:
`strtoint32()` inits `errno` before calling `strtoll()`. But we can keep it...

Reply:
fixed